Source/WebCore/svg/SVGElement.cpp

```diff
@@ -1034,7 +1037,11 @@ void SVGElement::svgAttributeChanged(const QualifiedName& attrName)
 Node::InsertedIntoAncestorResult SVGElement::insertedIntoAncestor(InsertionType insertionType, ContainerNode& parentOfInsertedTree)
 {
     StyledElement::insertedIntoAncestor(insertionType, parentOfInsertedTree);
-    updateRelativeLengthsInformation();
+
+    if (!m_hasInitializedRelativeLengthsState)
+        updateRelativeLengthsInformation();
+    else if (RefPtr parentElement = dynamicDowncast<SVGElement>(parentNode()); parentElement && &parentOfInsertedTree == parentNode())
```

Comment:
If the parent is an `SVGElement`, when is `&parentOfInsertedTree == parentNode()` not true here?

Reply:
parentOfInsertedTree is the parent of the node that got inserted into the container node. "This" node could be a descendant of such a node in which case this equality will evaluate to false.